desktop-ui/presentation/presentation.cpp

```diff
@@ -2,6 +2,12 @@
 namespace Instances { Instance<Presentation> presentation; }
 Presentation& presentation = Instances::presentation();
 
+#if defined(PLATFORM_MACOS)
+#define ELLIPSIS "…"
```

Comment:
In Python, if you use non-ASCII characters you need an encoding declaration to specify what encoding to use. I'd like to think that all compilers on macOS would assume (or even require UTF-8), but because of C's history it wouldn't surprise me if there's some circumstance where it's actually MacRoman or Windows1252 or something. What are the risks here?

Reply:
I strongly suggest using escape codes here instead of encoding declarations. `\u2026` for anything Unicode, which I would assume is the correct choice regardless and should automatically convert down to the correct format, `\x85` for Win-1252, `\xC9` for Mac Roman.

Reply:
Note that as long as the source file is stored with the correct encoding (recommended to be utf8) unicode should work everywhere.

Windows code should be doing utf8->utf16 conversion before calling WinAPIs with strings, when necessary

Reply:
C and Objective-C strings, file names, text files and so on are effectively guaranteed to be UTF-8 in macOS unless explicitly specified otherwise, in additional to what @LukeUsher already said. That said, I agree with Kawa's suggestion to use `\u2026` to avoid having non-ANSI characters in the file to begin with (so any accidental encoding change done on some other platform won't break macOS without anyone realizing).